egs/wsj/s5/steps/nnet3/convert_nnet2_to_nnet3.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/env python
+# Copyright 2016
+
+# This script converts nnet2 models into nnet3 models.
+# It requires knowledge of valid components which
+# can be modified in the configuration section below.
+
+import argparse, subprocess, os, tempfile, logging, sys, shutil, fileinput, re
+from collections import defaultdict, namedtuple
+import numpy as np
+
+# Begin configuration section
+# Components and their corresponding node names
+
+NODE_NAMES = {
+    "<AffineComponent>":"affine",
+    "<AffineComponentPreconditioned>":"affine",
+    "<AffineComponentPreconditionedOnline>":"affine",
+    "<BlockAffineComponent>":"affine",
+    "<BlockAffineComponentPreconditioned>":"affine",
+    "<SigmoidComponent>":"nonlin",
+    "<TanhComponent>":"nonlin",
+    "<PowerComponent>":"nonlin",
+    "<RectifiedLinearComponent>":"nonlin",
+    "<SoftHingeComponent>":"nonlin",
+    "<PnormComponent>":"nonlin",
+    "<NormalizeComponent>":"renorm",
+    "<MaxoutComponent>":"maxout",
+    "<MaxpoolingComponent>":"maxpool",
+    "<ScaleComponent>":"rescale",
+    "<DropoutComponent>":"dropout",
+    "<SoftmaxComponent>":"softmax",
+    "<LogSoftmaxComponent>":"log-softmax",
+    "<FixedScaleComponent>":"fixed-scale",
+    "<FixedAffineComponent>":"fixed-affine",
+    "<FixedLinearComponent>":"fixed-linear",
+    "<FixedBiasComponent>":"fixed-bias",
+    "<PermuteComponent>":"permute",
+    "<AdditiveNoiseComponent>":"noise",
+    "<Convolutional1dComponent>":"conv",
+    "<SumGroupComponent>":"sum-group",
+    "<DctComponent>":"dct",
+    "<SpliceComponent>":"splice",
+    "<SpliceMaxComponent>":"splice"
+}
+
+SPLICE_COMPONENTS = [c for c in NODE_NAMES if "Splice" in c]
+AFFINE_COMPONENTS = [c for c in NODE_NAMES if "Affine" in c]
+
+KNOWN_COMPONENTS = NODE_NAMES.keys()
+# End configuration section
+
+logger = logging.getLogger(__name__)
```

Comment:
I am not familiar with the latest logging guidelines. @vimalmanohar could you take a look.